File: backend/src/main/java/com/subastas/tpi/service/AuthService.java
package com.subastas.tpi.service;

import com.subastas.tpi.dto.request.LoginRequest;
import com.subastas.tpi.dto.request.RegisterRequest;
import com.subastas.tpi.dto.response.AuthResponse;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

public interface AuthService {

    void register(RegisterRequest request);

    AuthResponse login(LoginRequest request, HttpServletResponse response);

    AuthResponse refresh(HttpServletRequest request, HttpServletResponse response);

    void logout(HttpServletRequest request, HttpServletResponse response);
}

File: frontend/src/services/authService.ts
import api from './api';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface RegisterRequest {
  nombre: string;
  email: string;
  password: string;
  telefono: string;
}

export interface AuthResponse {
  accessToken: string;
  email: string;
  nombre: string;
}

export async function login(data: LoginRequest): Promise<AuthResponse> {
  const response = await api.post<AuthResponse>('/api/auth/login', data);
  return response.data;
}

export async function register(data: RegisterRequest): Promise<void> {
  await api.post('/api/auth/register', data);
}

export async function refresh(): Promise<AuthResponse> {
  const response = await api.post<AuthResponse>('/api/auth/refresh');
  return response.data;
}

export async function logout(): Promise<void> {
  await api.post('/api/auth/logout');
  localStorage.removeItem('access_token');
}
